examples/fit_ODNP_data.py

```diff
@@ -0,0 +1,99 @@
+"""Fitting ODNP Datasets for Ksigma
+===================================
+The T1(p) and E(p) integrals are pulled from an H5 file and are fit to extract the cross relaxivity of the sample. The R1(p) is first interpolated by fitting the self relaxivity to a polynomial. This fit is then used as a parameter for fitting the linear regime of the enhancement data that is extrapolated to the highest power. The cross relaxivity is then calculated using the output of these fits as well as the sample parameters that are fed to it (e.g., ppt value, and concentration).
+"""
+from pyspecdata import *
+from sympy import symbols, Symbol, latex
+from scipy.io import loadmat
+
+target_directory = os.path.normpath(getDATADIR("AG_processed_data"))
+h5_folder = "ras.h5"
+nodename = "220616_E37"
+# {{{Dataset Parameters
+SL_conc_M = 455.2223e-6
+ppt = 1.5167e-3
+Ep_pts = 18
+# }}}
+# {{{ plotting fn
+def list_symbs(f):
+    # {{{ this is just to show all the parameters
+    list_symbs = []
+    for j, k in f.output().items():
+        s_repr = latex(Symbol(j))
+        list_symbs.append(f"${s_repr} = {k:0.5g}$")
+    list_symbs = "\n".join(list_symbs)
+    # }}}
+    return list_symbs
+
+
+# }}}
+with figlist_var() as fl:
+    # {{{ load data
+    Ep = nddata_hdf5(f"{h5_folder}/{nodename}/Ep", directory=target_directory)
+    T1p = nddata_hdf5(f"{h5_folder}/{nodename}/T1p", directory=target_directory)
+    # }}}
+    # {{{Plot Ep
+    fl.next("E(p)")
+    fl.plot(Ep["power", :-3], "o", label="Experimental Data", capsize=6, alpha=0.5)
+    fl.plot(Ep["power", -3:], "rx", label="Returning Power Check", capsize=6, alpha=0.5)
+    plt.ylabel("E(p)")
+    # }}}
+    # {{{ plot T1p data
+    R1p = 1 / T1p
+    fl.next(r"$R_{1}(p)$")
+    fl.plot(R1p, "o", label="Experimental Data")
+    # }}}
+    # {{{load in T100 dataset
+    T10_p = loadmat('T10_DI_water_230412')['a'][0,:]
+    R10_p = nddata((T10_p[0]+T10_p[1]*R1p.getaxis('power'))**-1,'power')
+    R10_p.setaxis('power',R1p.getaxis('power'))
+    #}}}
+    #{{{ fit krho and R1p
+    krho_inv = SL_conc_M/(R1p - R10_p)
+    krho_inv_fit = krho_inv.polyfit('power',order=2)
+    krho_inv_fine = R1p.fromaxis('power').eval_poly(krho_inv_fit,'power')
+    R1p_fit = (R10_p + SL_conc_M/krho_inv_fine)
+    fl.plot(R1p_fit, ls=":", color="k", label="Fit", alpha=0.5)
+    plt.ylabel(r"$R_{1} / s^{-1}$")
+    plt.xlabel("Power / W")
+    # }}}
+    # {{{ Fit E(p)
+    fl.next("E(p)")
+    M0, A, phalf, p = symbols("M0 A phalf power", real=True)
+    sp = p / (p + phalf)
+    R1p = ((T10_p[0]+T10_p[1]*p)**-1 + (
+        SL_conc_M / (krho_inv_fit[0] + krho_inv_fit[1] * p + krho_inv_fit[2] * p**2)))
```

Comment:
why can't this just be `R1p_fit` -- alternately, why can't this be shown on the R1(p) plot?

Reply:
the sympy expression (which R1p is used in) must be symbolic 
`AssertionError: for now, the functional form must be a sympy expression`

Reply:
But edited the R1(p) fit plot to use this expression instead